src/GeekLearning.Domain/Validation/ValidatableAggregateBase.cs

```diff
@@ -44,7 +45,9 @@ public Task<IValidationResult> ValidateAsync()
     public abstract class ValidatableAggregateBase<TEntity, TValidator> : AggregateBase<TEntity>, IValidatableAggregate
        where TValidator : IValidator
     {
-        protected abstract IValidator validator { get; }
+        private IValidator validator;
+
+        protected virtual IValidator WithValidator(IValidatorFactory factory) => validator = factory.GetValidator<TValidator>();
```

Comment:
Maybe we should offer both ways to register the IValidatorFactory (through this way, but also with another constructor)?